src/helpers/image-helper.php

```diff
@@ -61,14 +100,27 @@ public function __construct(
 		$this->indexable_repository = $indexable_repository;
 		$this->options_helper       = $options;
 		$this->url_helper           = $url_helper;
+
+		$this->home_url = \home_url();
+		$parsed_home    = \wp_parse_url( $this->home_url );
+
+		if ( ! empty( $parsed_home['host'] ) ) {
+			$this->host = \str_replace( 'www.', '', $parsed_home['host'] );
+		}
+
+		if ( ! empty( $parsed_home['scheme'] ) ) {
+			$this->scheme = $parsed_home['scheme'];
+		}
```

Comment:
All of the above logic feels like it should live either in a separate helper or higher order somewhere.